core_v14/src/decoder/extrinsic_bytes.rs

```diff
@@ -47,30 +47,42 @@ impl<'a> AllExtrinsicBytes<'a> {
 	/// with each one (not including the length prefix), or an error containing the position
 	/// at which decoding failed.
 	pub fn iter(&self) -> ExtrinsicBytesIter<'a> {
-		ExtrinsicBytesIter { data: self.data, cursor: 0 }
+		ExtrinsicBytesIter { remaining_count: self.len, data: self.data, cursor: 0 }
 	}
 }
 
 /// An iterator that returns the set of bytes representing each extrinsic found.
 /// On each iteration, we return either the extrinsic bytes, or an error containing
 /// the position at which decoding failed.
 pub struct ExtrinsicBytesIter<'a> {
+	remaining_count: usize,
```

Comment:
I was curious about this too.

Reply:
We know how many extrinsics we expect to see (the original length). This keeps track of how many extrinsics (not bytes; I can see now that the name isn't clear!) are expected to be remaining, so that we stop iteration once the number hits 0 and don't try to consume more bytes than we need to (so that we can return the rest afterwards).

Reply:
ah, I see now. I totally thought this was bytes. Fair enough.